include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -289,7 +289,45 @@ namespace LinearAlgebra
       //  - Third case: the vectors have different size.
       if (vector->getMap()->isSameAs(*V.vector->getMap()))
         {
-          *vector = *V.vector;
+          // Create a read-only Kokkos view from the source vector
+#  if DEAL_II_TRILINOS_VERSION_GTE(13, 2, 0)
+          auto source_vector_2d =
+            V.vector->template getLocalView<Kokkos::HostSpace>(
+              Tpetra::Access::ReadOnly);
+#  else
+          auto source_vector_2d =
+            V.vector->template getLocalView<Kokkos::HostSpace>();
+#  endif
+          auto source_vector_1d =
+            Kokkos::subview(source_vector_2d, Kokkos::ALL(), 0);
+
+          // Create a read/write Kokkos view from the target vector
+#  if DEAL_II_TRILINOS_VERSION_GTE(13, 2, 0)
+          auto target_vector_2d =
+            vector->template getLocalView<Kokkos::HostSpace>(
+              Tpetra::Access::ReadWrite);
+#  else
+          vector->template sync<Kokkos::HostSpace>();
+          auto target_vector_2d =
+            vector->template getLocalView<Kokkos::HostSpace>();
+#  endif
+          auto target_vector_1d =
+            Kokkos::subview(target_vector_2d, Kokkos::ALL(), 0);
+#  if !DEAL_II_TRILINOS_VERSION_GTE(13, 2, 0)
+          vector->template modify<Kokkos::HostSpace>();
+#  endif
+
+          // Copy the data
+          const size_t local_length = vector->getLocalLength();
+          for (size_t i = 0; i < local_length; ++i)
+            target_vector_1d(i) = source_vector_1d(i);
+
```

Comment:
Why not just
```suggestion
          Kokkos::deep_copy(target_vector_1d, source_vector_1d);
```
?

Reply:
I did not know that function.
But that is a good remark!